nixos/modules/programs/steam.nix

```diff
@@ -191,7 +191,10 @@ in {
       };
     };
 
-    programs.steam.extraPackages = cfg.fontPackages;
+    programs.steam.extraPackages = [ (pkgs.runCommand "X11-fonts" { preferLocalBuild = true; } ''
+      mkdir -p "$out/share"
+      ln -s "${pkgs.symlinkJoin { name = "steam-fonts"; paths = cfg.fontPackages; }}/share/fonts" "$out/share/fonts"
+    '')];
```

Comment:
It is enough to just change the type of `extraPackages` and `fontPackages` to `types.listOf types.path`.  After skimming through codes related to steam, buildFHSEnv and [buildEnv][1], I find that the `package` type is stronger than needed and the `path` type can be handled.

[1]: https://github.com/NixOS/nixpkgs/blob/2cd4abd34d2eddacb62f60a4c157a0a0c8d1817d/pkgs/build-support/buildenv/default.nix#L58-L72

Reply:
The disadvantage of this proposal is that if an element of `extraPackages` or `fontPackages` is not a `package` (derivation) it will be linked to a wrong place so it won't be available in the steam environment.

Reply:
What I'm more curious about is whether there are other practical use cases where fonts.packages is a path. For `ghostscript` this is irrelevant, since `/usr/share/ghostscript/fonts` is not in steam's search path.

Reply:
BTW, just note that your original implementation also has this disadvantage.